Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -178,18 +178,30 @@ private static IEnumerable<BusinessObjectValidationFailure> CreateBusinessObject
       return result;
     }
 
-    public IReadOnlyCollection<UnhandledBusinessObjectValidationFailure> GetUnhandledValidationFailures (
-        IBusinessObject businessObject,
-        bool includePartiallyHandledFailures = false)
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailures (
+        [NotNull] IBusinessObject businessObject,
+        bool includePartiallyHandledFailures = false,
+        bool markAsHandled = false)
     {
-      ArgumentUtility.CheckNotNull("businessObject", businessObject);
+      ArgumentUtility.CheckNotNull(nameof(businessObject), businessObject);
 
-      return _unhandledValidationFailures
+      var validationFailures = _validationFailures
           .Where(f => Equals(f.BusinessObjectValidationFailure.ValidatedObject, businessObject))
-          .Where(f => includePartiallyHandledFailures || !_handledValidationFailures.Contains(f.ValidationFailure))
-          .Select(f => f.BusinessObjectValidationFailure)
-          .Select(f => new UnhandledBusinessObjectValidationFailure(f.ErrorMessage, f.ValidatedProperty))
-          .ToArray();
+          .Where(f => includePartiallyHandledFailures || !_handledValidationFailures.Contains(f.ValidationFailure));
+
+      var result = new List<BusinessObjectValidationFailure>();
+      foreach (var (businessObjectValidationFailure, validationFailure) in validationFailures)
+      {
+        result.Add(businessObjectValidationFailure);
+
+        if (markAsHandled)
+        {
+          _unhandledValidationFailures.Remove((businessObjectValidationFailure, validationFailure));
+          _handledValidationFailures.Add(validationFailure);
```

Comment:
Not sure, should we remove the code duplication that stems from these two lines (they are also used in GetValidationFailuers)